### PR TITLE
Update openshift documentation, manifests for metricbeat and elastic-agent standalone

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Signed-off-by: Tetiana Kravchenko <tetiana.kravchenko@elastic.co>

## What does this PR do?
- metricbeat documentation: add kubernetes.proxy config for openshift setup
- add comments related to the openshift installation for metricbeat and elastic-agent standalone

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues
- Related https://github.com/elastic/integrations/issues/2065
- Closes https://github.com/elastic/beats/issues/17863

### CODE REVIEW

File: metricbeat/docs/running-on-kubernetes.asciidoc [hunk 1]
```diff
@@ -88,16 +88,7 @@ spec:
 ===== Red Hat OpenShift configuration
 
 If you are using Red Hat OpenShift, you need to specify additional settings in
-the manifest file and enable the container to run as privileged.
```
Comment (ChrsMark) [inline metricbeat/docs/running-on-kubernetes.asciidoc:91]: Are we removing this?

File: metricbeat/docs/running-on-kubernetes.asciidoc [hunk 2]
```diff
@@ -118,6 +109,12 @@ specify the following settings under `kubernetes.yml` in the `data` section:
       bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
       ssl.certificate_authorities:
         - /path/to/kubelet-service-ca.crt
+    - module: kubernetes
+      metricsets:
+        - proxy
+      period: 10s
+      host: ${NODE_NAME}
+      hosts: ["localhost:29101"]
```
Comment (ChrsMark) [inline metricbeat/docs/running-on-kubernetes.asciidoc:117]: Is the only thing needed? Any notes or links to how the proxy should be tuned on k8s' side?

Comment (tetianakravchenko) [reply metricbeat/docs/running-on-kubernetes.asciidoc:117]: There is no need to change `sdn-config` configMap, this is the default configuration:
```
kube-proxy-config.yaml: |-
    apiVersion: kubeproxy.config.k8s.io/v1alpha1
    bindAddress: 0.0.0.0
    bindAddressHardFail: false
    clientConnection:
      acceptContentTypes: ""
      burst: 0
      contentType: ""
      kubeconfig: ""
      qps: 0
    clusterCIDR: 10.128.0.0/14
    configSyncPeriod: 0s
    conntrack:
      maxPerCore: null
      min: null
      tcpCloseWaitTimeout: null
      tcpEstablishedTimeout: null
    detectLocalMode: ""
    enableProfiling: true
    healthzBindAddress: 0.0.0.0:10256
    hostnameOverride: ""
    iptables:
      masqueradeAll: false
      masqueradeBit: 0
      minSyncPeriod: 0s
      syncPeriod: 0s
    ipvs:
      excludeCIDRs: null
      minSyncPeriod: 0s
      scheduler: ""
      strictARP: false
      syncPeriod: 0s
      tcpFinTimeout: 0s
      tcpTimeout: 0s
      udpTimeout: 0s
    kind: KubeProxyConfiguration
    metricsBindAddress: 0.0.0.0:29101
    mode: unidling+iptables
    nodePortAddresses: null
    oomScoreAdj: null
    portRange: ""
    showHiddenMetricsForVersion: ""
    udpIdleTimeout: 0s
    winkernel:
      enableDSR: false
      networkName: ""
      sourceVip: ""
``` 
note here: `metricsBindAddress: 0.0.0.0:29101`

File: metricbeat/docs/running-on-kubernetes.asciidoc (hunk 1 above)
Comment (tetianakravchenko) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: I've checked it on Openshiftv4 (GCP setup) and with `privileged: true` and without `privileged: true` getting the same result:
from the container:
```
mount | grep /usr/share/metricbeat/data
/dev/sda4 on /usr/share/metricbeat/data type xfs (rw,relatime,seclabel,attr2,inode64,logbufs=8,logbsize=32k,prjquota)
```
no errors descibed here: https://github.com/elastic/beats/issues/17516#issue-595098235

additionally to that, I wanted to keep metricbeat and elastic-agent manifests (beside the actual configuration of modules/datasets) the same - https://github.com/elastic/observability-docs/pull/1498

File: metricbeat/docs/running-on-kubernetes.asciidoc (hunk 2 above)
Comment (ChrsMark) [reply metricbeat/docs/running-on-kubernetes.asciidoc:117]: Nice!

File: metricbeat/docs/running-on-kubernetes.asciidoc (hunk 1 above)
Comment (ChrsMark) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: That's good! However maybe we need to be more explicit here and mention in which versions it is not required to avoid confusion?

Comment (tetianakravchenko) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: Should we rely on openshift version here instead of kubernetes version? according to this [doc](https://docs.openshift.com/container-platform/3.11/release_notes/ocp_3_11_release_notes.html#ocp-311-about-this-release) the last available openshift version 3.11 uses Kubernetes 1.11, which is not supported by us I think

Comment (ChrsMark) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: Strictly speaking yes we can claim that older versions are not supported however a small comment like `NOTE: for older versions of v4 users need to run the container as privileged` so as to provide a good hint to possible users that are still with older versions.

Comment (tetianakravchenko) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: I will add it.
I didn't check the metricbeat on openshift v3, but somehow I think it even might not work (at least some functionality) - issue described in this comment https://github.com/elastic/integrations/issues/2065#issuecomment-1002215969

File: metricbeat/metricbeat.yml [hunk 3]
```diff
@@ -60,16 +60,16 @@ setup.template.settings:
 # This requires a Kibana endpoint configuration.
 setup.kibana:
 
-# Kibana Host
-# Scheme and port can be left out and will be set to the default (http and 5601)
-# In case you specify and additional path, the scheme is required: http://localhost:5601/path
-# IPv6 addresses should always be defined as: https://[2001:db8::1]:5601
-#host: "localhost:5601"
+  # Kibana Host
```
Comment (ChrsMark) [inline metricbeat/metricbeat.yml:63]: Is this related?

Comment (tetianakravchenko) [reply metricbeat/metricbeat.yml:63]: no, it is a result of `make update` in other case tests were failing 🤷‍♀️

File: metricbeat/docs/running-on-kubernetes.asciidoc (hunk 1 above)
Comment (tetianakravchenko) [reply metricbeat/docs/running-on-kubernetes.asciidoc:91]: done - https://github.com/elastic/beats/pull/30054/commits/8d1e5d25e17794c52e854ad1c4b0eb7921d00860

**Review by ChrsMark — APPROVED** [review]
